src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -306,6 +381,9 @@ def validate_metadata(
         lockfile=lockfile,
         user_interpreter_constraints=interpreter_constraints,
         user_requirements=user_requirements,
+        maybe_constraints_file_path=constraints_file_path_and_hash[0]
```

Comment:
Types certainly mean 0 must be the correct index, but its definitely not reader-friendly.

Reply:
Agreed. I considered using `NamedTuple`, but felt confusing to introduce yet another class when we already have so many. So I was hoping the `path_and_hash[0]` would be clear enough it means `path`. Thoughts?

Reply:
I think I expressed my thoughts! I'd personally break out a 2 field dataclass, but I ship-it-ed because this is all fashion and not too important to the end goal.

Reply:
I second that this should be a dataclass.